Migrate remaining providers to common.compat compatibility layer in celery

rich7420 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->
Replace version-specific conditional imports with common.compat layer. This standardizes compatibility handling across Airflow 2.x and 3.x. 

This PR is a part of [57018](https://github.com/apache/airflow/issues/57018) about provider celery.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

rich7420 committed:
Migrate remaining providers to common.compat compatibility layer

rich7420 committed:
Add TODO for version bump in optional dependencies

Added a TODO comment to indicate the need to bump the version of 'apache-airflow-providers-common-compat'.

rich7420 committed:
fix ci test error

bugraoz93 reviewed (commented):
Looks good overall! Could you please check the CI, the static checks failing. The Airflow CTL integration tests are fixed in main, rebase should fix that

rich7420 committed:
Fix ruff linting error: move timeout import to top of file

- Move 'from airflow.providers.common.compat.sdk import timeout' to top of file
- Fixes E402 Module level import not at top of file error

rich7420 committed:
fix static check error

eladkal commented on providers/celery/pyproject.toml:
```diff
@@ -69,6 +69,9 @@ dependencies = [
 [project.optional-dependencies]
+"common.compat" = [
+    "apache-airflow-providers-common-compat>=1.7.4",  # + TODO: bump to next version
+]
```
Why as optional dependency?

rich7420 replied:
You're right , my mistake.

rich7420 committed:
add hash

amoghrajesh commented on providers/celery/pyproject.toml:
```diff
@@ -69,6 +69,9 @@ dependencies = [
 [project.optional-dependencies]
+"common.compat" = [
+    "apache-airflow-providers-common-compat>=1.7.4",  # + TODO: bump to next version
+]
```
This has to go to "dependencies"?

rich7420 replied:
@amoghrajesh Yes! You're right. Thanks for your feedback!

rich7420 committed:
update for fixing

rich7420 committed:
Update hash value in www-hash.txt

kaxil commented on providers/celery/pyproject.toml:
```diff
@@ -58,6 +58,7 @@ requires-python = ">=3.10"
 # After you modify the dependencies, and rebuild your Breeze CI image with ``breeze ci-image build``
 dependencies = [
     "apache-airflow>=2.10.0",
+    "apache-airflow-providers-common-compat>=1.8.0",  # + TODO: bump to next version
```
```suggestion
    "apache-airflow-providers-common-compat>=1.8.0",
```

rich7420 committed:
Update pyproject.toml

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

rich7420 committed:
test hash
